Sources/BraveWallet/Crypto/Stores/BuyTokenStore.swift

```diff
@@ -152,6 +176,7 @@ public class BuyTokenStore: ObservableObject {
     
     let coin = await walletService.selectedCoin()
     selectedNetwork = await rpcService.network(coin)
+    await validatePrefilledToken(on: selectedNetwork) // selectedNetwork may change
```

Comment:
when there is no `prefilledToken` we dont need to validate the current selected network

Reply:
The function returns immediately if prefilledToken is nil